### ISSUE TITLE
Disable WebExtensions in Basilisk

### ISSUE THREAD

**Issue by wolfbeast** [MoonchildProductions/UXP#937]
Blocks #936 

This requires building with `--disable-webextensions` and testing functionality in Basilisk, and fixing any issues that may arise.

**Comment by mattatobin** [comment]
http://xref.palemoon.org/uxp-trunk/source/application/basilisk/confvars.sh#58

**Comment by wolfbeast** [comment]
Tested with a webextension (bookmarksorganizer) installed, then upgrading the profile to a version without WE support.

Popped up the "checking your add-ons" wizard and it hung there, pressing cancel was the only way out.

Error console:
```
1548031951122	addons.xpi	WARN	updateMetadata: Add-on bookmarksorganizer@agenedia.com is invalid: {"name":"UnsupportedExtension","message":"Basilisk does not support WebExtensions","webext":true}
1548032027372	addons.manager	WARN	Exception calling callback: TypeError: addon is null (file:///e:/mozdev/UXP/obj-x86-unstable/dist/bin/browser/components/nsBrowserGlue.js:1113:1) JS Stack trace: BG__onWindowsRestored/<@nsBrowserGlue.js:1113:1 < safeCall@AddonManager.jsm:165:5 < noMoreObjects@AddonManager.jsm:2154:9 < callNext@AddonManager.jsm:328:7 < getAddonsByIDs_getAddonByID@AddonManager.jsm:2149:11 < safeCall@AddonManager.jsm:165:5 < noMoreObjects@AddonManager.jsm:2075:9 < callNext@AddonManager.jsm:328:7 < getAddonByID_safeCall@AddonManager.jsm:2070:13 < PL_getAddon@PluginProvider.jsm:135:7 < callProviderAsync@AddonManager.jsm:227:12 < nextObject@AddonManager.jsm:2065:9 < callNext@AddonManager.jsm:334:7 < getAddonByID_safeCall@AddonManager.jsm:2070:13 < getAddonByID@GMPProvider.jsm:542:7 < callProviderAsync@AddonManager.jsm:227:12 < nextObject@AddonManager.jsm:2065:9 < callNext@AddonManager.jsm:334:7 < getAddonByID_safeCall@AddonManager.jsm:2070:13 < getAddonByID@LightweightThemeManager.jsm:362:7 < callProviderAsync@AddonManager.jsm:227:12 < nextObject@AddonManager.jsm:2065:9 < callNext@AddonManager.jsm:334:7 < getAddonByID_safeCall@AddonManager.jsm:2070:13 < getAddonByID_getVisibleAddonForID@XPIProvider.jsm:3809:7 < makeSafe/<@XPIProviderUtils.js:147:17 < getRepositoryAddon@XPIProviderUtils.js:128:5 < getAddon/<@XPIProviderUtils.js:1076:9 < process@Promise-backend.js:932:23 < walkerLoop@Promise-backend.js:813:7 < scheduleWalkerLoop/<@Promise-backend.js:747:11
```

**Comment by wolfbeast** [comment]
Good news: invalid webextensions after the hung check are automatically purged from the extension store in the profile. We don't have to do anything for it; it will automatically be cleaned up from the profile.

**Comment by tauxeta** [comment]
@wolfbeast @mattatobin 

Someone needs to respond to these accusation otherwise you're letting more misinformation spread out there about Pale moon, yeah this guy is gorhill, the uBO guy.
![gorhill](https://user-images.githubusercontent.com/8204777/51522599-67c9fe80-1e32-11e9-8ba6-0aca9e6cbfc8.PNG)
https://www.reddit.com/r/firefox/comments/ai95us/basilisk_browser_drops_webextension_support/een8fr0/

**Comment by wolfbeast** [comment]
GMP needs to be repaired with the changeover of add-on manager so EME still works.

**Comment by mattatobin** [comment]
@trav90 Would you please build Basilisk from the `TychoEME-work` branch and check to see if the netflix works?

**Comment by trav90** [comment]
I don’t currently have access to a Netflix account but I will build and can test EME playback using YouTube’s test suite.

**Comment by wolfbeast** [comment]
Netflix is officially not supporting Linux anyway

**Comment by trav90** [comment]
EME playback appears to work, using YouTube's [EME conformance tests](https://yt-dash-mse-test.commondatastorage.googleapis.com/unit-tests/2018.html?test_type=encryptedmedia-test&timestamp=1548850239638). The browser fails the tests themselves, but so does current Basilisk-release so that is caused by something else.

**Comment by mattatobin** [comment]
Thank you @trav90

**Comment by wolfbeast** [comment]
I'll be merging this in and do some test runs.

**Comment by mattatobin** [comment]
Okay, this has little to do with this specific issue.. This has to do with switching the Add-ons Manager to the Tycho-sourced version and as part of that to make sure the CDMs download and install properly.. Not WHICH one does but that it DOES happen.

Your entire post while helpful is very verbose and it was difficult to pull out what was nessisary..

tl;dr we need to change the AUS pref to get what is delivered for `60.4` to get the new CDM version because the older one is being withdrawn by the Google. Also test to make sure it still works with our EME and GMP support code.

Someone do that then.

**Comment by Vangelis66** [comment]
> this has little to do with this specific issue

Please accept my most sincere apologies; I just felt an impulse to share additional info wrt what @trav90 had hinted... If a separate issue has to be started for WV breakage, I take it one of the team will kindly do so...

> we need to change the AUS pref to get what is delivered for `60.4`

... or possibly higher, to `60.5.0` (which will fetch WV v4.10.1196.0)

> it still works with our EME (snipped) support code. 

That would be the most difficult part, IMHO... As a non-coder, I don't have anything additional to contribute, so thanks for your patience and time, I'll let the pros take over 😄

**Comment by mattatobin** [comment]
No the info is very helpful and appreciated but I was a bit lost on extracting the action points.

**Comment by wolfbeast** [comment]
And at the same time Netflix worked fine, just to confirm, from the current tip of our trunk.
Of note we need to stop relying on aus*.mozilla.org and host the update xml ourselves.

**Comment by mattatobin** [comment]
Do the new CDMs work?

**Comment by Vangelis66** [comment]
Hello again 😄 
I have already posted official, business grade, WidevineCDM tests previously in this thread: 
 
1. https://bitmovin.com/demos/drm
2. https://drm.axinom.com/bitmovin-and-unified-streaming/
3. https://demo.castlabs.com/ (Tests 3+4, with DRM)
4. https://shaka-player-demo.appspot.com/demo/#asset=https://storage.googleapis.com/shaka-demo-assets/dig-the-uke/dash.mpd;lang=en-US;logtoscreen;build=debug_compiled

and these don't work here, either with 1.4.9.1088 or 4.10.1196.0 (see #956); as I already said,  

>> it still works with our EME (snipped) support code.
> 
> That would be the most difficult part, IMHO...

Netflix has a tendency to silently fall back to Silverlight NPAPI when the WV implementation is not-working and when Silverlight is installed and enabled (and that's the reason Netflix can work in Pale Moon with a SSUAO+Silverlight); I have not myself a Netflix subscription, some clips/trailers there are enencrypted, unlike full episodes, so I humbly advise to use linked tests for good measure... 

Furthermore, I have conducted myself Widevine related searches in Bugzilla, the results of which I hesitated posting either here or in #956 - being OT): 
1. In any given time, supported widevine versions are indicated in 
https://dl.google.com/widevine-cdm/versions.txt
Right now, these are: 
 
(interface 9)
1.4.**9**.1088
(interface 10)
4.**10**.1146.0
4.**10**.1192.0
4.**10**.1196.0

2. It would take a lot of work to upgrade Basilisk code from interface **8** Widevine compatibility (by default v1.4.**8**.903, then v1.4.**8**.1088) to first interface **9** and then **10** Widevine compatibility; several bugzilla bugs refer to that, public ones I had access to are: 

https://bugzilla.mozilla.org/show_bug.cgi?id=1413835
https://bugzilla.mozilla.org/show_bug.cgi?id=1420836
https://bugzilla.mozilla.org/show_bug.cgi?id=1475260
https://bugzilla.mozilla.org/show_bug.cgi?id=1475277
https://bugzilla.mozilla.org/show_bug.cgi?id=1486482
https://bugzilla.mozilla.org/show_bug.cgi?id=1486502
https://bugzilla.mozilla.org/show_bug.cgi?id=1487416
https://bugzilla.mozilla.org/show_bug.cgi?id=1487811
https://bugzilla.mozilla.org/show_bug.cgi?id=1491117
https://bugzilla.mozilla.org/show_bug.cgi?id=1491889
https://bugzilla.mozilla.org/show_bug.cgi?id=1492377
https://bugzilla.mozilla.org/show_bug.cgi?id=1493255
https://bugzilla.mozilla.org/show_bug.cgi?id=1494178
https://bugzilla.mozilla.org/show_bug.cgi?id=1495514
https://bugzilla.mozilla.org/show_bug.cgi?id=1496501
https://bugzilla.mozilla.org/show_bug.cgi?id=1499880
https://bugzilla.mozilla.org/show_bug.cgi?id=1503124
https://bugzilla.mozilla.org/show_bug.cgi?id=1514062

Some of them might not be 100% relevant to UXP, just posting for simple reference (?)...

Also of possible interest: 

https://github.com/mozilla-releng/releasewarrior-2.0/blob/master/docs/misc-operations/widevine-updates.md

I have a vested interest in having Basilisk work with latest WidevineCDM, as I'd hate to have to use Quantum and, God forbid, Google Chrome and forks for Widevine sites (e.g. Spotify web player); Pale Moon serves me right currently, Basilisk is used for instances where PM is not sufficient 😉
And although I might be more verbose than probably allowed, I post simply to help and arm the devs with knowledge (what little I possess...) 

Thanks again to the whole Moonchild team for what they offer!

**Comment by trav90** [comment]
@Vangelis66 the majority of those bugs will not apply to UXP (most are nothing more than Mozilla updating their AUS rules and/or internal scripts to fetch new/different versions of Widevine), though there are a few that should be evaluated further. I will open a new issue for this so we can return this issue discussion to disabling WebExtensions.

**Comment by Vangelis66** [comment]
@trav90 wrote: 

> the majority of those bugs will not apply to UXP 
> (most are nothing more than Mozilla updating 
> their AUS rules and/or internal scripts 
> to fetch new/different versions of Widevine)

... Sure, but further down in the comments section for such bugs lie additional links for possibly UXP relevant WV bugs, hence why I posted those 😄 ...

> I will open a new issue for this (#962)

Many thanks; being pedantic, issue title should've read "... for Widevine **1**.4.9 & 4.10"

> so we can return this issue discussion to disabling WebExtensions.

Agreed 👍

**Comment by wolfbeast** [comment]
This has been effected and published. Closing.
